src/expressions/operator-expr.md

```diff
@@ -70,6 +70,50 @@ let a = && && mut 10;
 let a = & & & & mut 10;
 ```
 
+### Raw address-of operators
+
+Related to the borrow operators are the *raw address-of operators*, which do not have first-class syntax, but are exposed via the macros `ptr::addr_of!(expr)` and `ptr::addr_of_mut!(expr)`.
+The expression `expr` is evaluated in place expression context.
+`ptr::addr_of!(expr)` then creates a (const) raw pointer of type `*const T` to the given place, and `ptr::addr_of_mut!(expr)` creates a mutable raw pointer of type `*mut T`.
+
+The raw address-of operators must be used instead of a borrow operator whenever the place expression could evaluate to a place that is not properly aligned or does not store a valid value as determined by its type, or whenever creating a reference would introduce incorrect aliasing assumptions.
+In those situations, using a borrow operator would cause [undefined behavior] by creating an invalid reference, but a raw pointer may still be constructed using an address-of operator.
+
+The following is an example of creating a raw pointer to an unaligned place through a `packed` struct:
+
+```rust
+use std::ptr;
+
+#[repr(packed)]
```

Comment:
Does this maybe need to be a well-defined layout like C?  I would assume that the Rust layout could theoretically rearrange the order?

```suggestion
#[repr(C, packed)]
```

Reply:
> I would assume that the Rust layout could theoretically rearrange the order?

It could, but nothing in the example relies on field order. So this is correct without `C`, I think.
We could still add it though, if you want?

Reply:
Ah, makes sense. Leaving as-is seems fine to me.